### ISSUE TITLE
Product Button Block: Style duplicates

### ISSUE THREAD

**Issue by dashkevych** [woothemes/woocommerce#42307]
## Describe the bug

WooCommerce loads the same styles as WordPress core for product button (`woocommerce/product-button`) when Outline style is selected.

This styles are loaded via `wc-blocks-style.css`. Here are those styles: 

```css
.wp-block-button.is-style-outline .wp-block-button__link:not(.has-background) {
    background-color: transparent;
    background-image: none;
}

.wp-block-button.is-style-outline .wp-block-button__link:not(.has-text-color) {
    color: currentColor;
}

.wp-block-button.is-style-outline .wp-block-button__link {
    border: 2px solid;
}
```

These styles mimic styles for Button block which comes from a WordPress core.

When Button block (core) and Product Button block (woocommerce) are both used on the page then the page duplicates the styles.

Since these styles are related to a WordPress core, is it a responsibility of a core to load these styles?

## To reproduce

1. Create new page.
2. Add Button block (from WordPress core) with Outline style.
3. Add Products (Beta) block (from WooCommerce).
4. Select Product Button
5. Set Outline style for button.

## Expected behavior

Styles for button with Outline styles should be loaded only once and should come from WordPress core since these styles are part of a WordPress core.

## Screenshots

![Preview of CSS duplicates caused by WooCommerce](https://i.snipboard.io/zPJwDO.jpg)

### WordPress:

-   WordPress version: 6.2.2
-   WooCommerce version: 7.8.2
-   Any other plugins installed: Only WooCommerce

**Comment by gigitux** [comment]
Thanks for reporting this issue! I'm able to replicate it. The cause lies in the `product button` block is different from the `core/button` block, preventing consistent CSS reuse.

There is a dedicated issue on refactor the `product button` block: https://github.com/woocommerce/woocommerce/issues/53138

**Comment by gigitux** [comment]
I'm going to mark this issue as a duplicate of https://github.com/woocommerce/woocommerce/issues/56685
